File: Leetcode_Code/3. Longest Substring Without Repeating Characters/3.java
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxl = 0, start = 0;
        Map<Character, Integer> dict = new HashMap<>();
        for (int i = 0; i < s.length(); ++i) {
            if (dict.get(s.charAt(i)) == null || dict.get(s.charAt(i)) < start) {
                maxl = Math.max(maxl, i - start + 1);
            } else {
                start = dict.get(s.charAt(i)) + 1;
            }
            dict.put(s.charAt(i), i);
        }
        return maxl;
    }
}

File: Leetcode_Code/3. Longest Substring Without Repeating Characters/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxl = 0, start = 0;
        map<char, int> dict;
        for (int i = 0; i < s.length(); ++i) {
            if (dict[s[i]] == 0 || dict[s[i]] < start) {
                maxl = max(maxl, i - start + 1);
            } else {
                start = dict[s[i]];
            }
            dict[s[i]] = i + 1;
        }
        return maxl;
    }
};
